test/Microsoft.Dnx.Testing.CommonUtils/Directory/Dir.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class Dir
+    {
+        public const string EmptyFile = "";
+
+        private readonly Dictionary<string, object> _nodes = new Dictionary<string, object>();
+        private Func<FileInfo, object> _readFile;
+
+        public Dir(string rootPath): this()
+        {
+            if (!Directory.Exists(rootPath))
+            {
+                throw new DirectoryNotFoundException();
+            }
+
+            LoadPath = rootPath;
+            Load(rootPath);
+        }
+
+        public Dir()
+        {
+            _readFile = fileInfo => new Lazy<object>(() => File.ReadAllText(fileInfo.FullName));
```

Comment:
The lazy is an implementation  detail. Don't return it to the caller, just do the lazy part inside this object